init.cc

```diff
@@ -16,7 +16,9 @@
 
 logging::logger startlog("init");
 
-std::set<gms::inet_address> get_seeds_from_db_config(const db::config& cfg, gms::inet_address broadcast_address) {
+std::set<gms::inet_address> get_seeds_from_db_config(const db::config& cfg,
+                                                     const gms::inet_address broadcast_address,
+                                                     const bool fail_on_lookup_error /* = true */) {
```

Comment:
I don't like the `/* = true */` comment. Makes it easy to shoot yourself in the foot later if you modify the declaration site and forget about the comment, which would then cause confusion.

In fact I don't like that the parameter has a default value assigned at all -- I have debugged countless bugs coming from someone forgetting to assign a correct value to a parameter, because it had a default. I suggest removing it. There is exactly one call site for this function -- what's the point of the default?

Reply:
Ok, valid concern.  The reason of having a default is to keep the previously existing interface of that function.

Reply:
(I would mention that having non-const correct code also makes it much easier to shoot yourself in the foot)

Reply:
As for the by-value parameters, there is actually quite a discussion if const-ing them is good or bad - in particular, it brings some visual clutter, and when people see `const some_type name` they might be expecting that the argument should have been passed by a const ref instead, or even when it really should have been passed via a const ref it could be easily missed in the reviews that the ref is missing when there is a const in the front.

At least the `const` should not be used for the API by-value parameters (= declarations), which is good here, as you are only using them at the definition side. But e.g. here this argues even against that: https://quuxplusone.github.io/blog/2022/01/23/dont-const-all-the-things/

(in this case, looking just at this definition, I might be wondering if the `const gms::inet_address broadcast_address` shouldn't actually be `const gms::inet_address& broadcast_address` instead)

On the other side, it is indeed correct that the const for the by-value parameter will prevent changing it unintentionally inside the function (which will not have any external effect). But it seems to not be used heavily in the codebase right now.

Reply:
I'm not insisting that const correctness is a silver bullet and should be used everywhere. Of course there are cases when const should not be used. But I see no harm in marking even int vars as const if they are not mutated. 
Regarding pass-by value and pass-by reference, on 64 bit x86 cpus it's more effective to pass one-two word (<=16 bytes) objects by value and everything else by a const ref, if you are using a const ref for small objects and the function is not inlined, it will prevent the compiler from using registers for passing the value.